src/main/java/it/gov/pagopa/pu/bff/connector/registries/client/PagoPaRegistryClient.java

```diff
@@ -0,0 +1,27 @@
+package it.gov.pagopa.pu.bff.connector.registries.client;
+
+import it.gov.pagopa.pu.bff.connector.registries.config.RegistriesApisHolder;
+import it.gov.pagopa.pu.bff.exception.ResourceNotFoundException;
+import it.gov.pagopa.pu.registries.dto.generated.PagoPaRegistryDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+@Service
+@Slf4j
+public class PagoPaRegistryClient {
+  private final RegistriesApisHolder registriesApisHolder;
+
+  public PagoPaRegistryClient(RegistriesApisHolder registriesApisHolder) {
+    this.registriesApisHolder = registriesApisHolder;
+  }
+
+  public PagoPaRegistryDTO getPagoPaRegistry(String pagoPaRegistryId, String accessToken) {
+    try {
+      return registriesApisHolder.getPagoPaRegistryApi(accessToken)
+              .getPagoPaRegistry(pagoPaRegistryId);
+    } catch (HttpClientErrorException.NotFound e) {
+      throw new ResourceNotFoundException("PagoPaRegistry with ID "+pagoPaRegistryId+" not found");
```

Comment:
404 == null
we have always returned null client side... if it's not supposed to be null, usually we perform this check and throw the error on business logic